src/main/java/bio/terra/service/snapshotbuilder/utils/HierarchyQueryBuilder.java

```diff
@@ -75,15 +87,51 @@ private static BinaryFilterVariable requireStandardConcept(TableVariable concept
    * </pre>
    */
   private Query selectAllParents(int conceptId) {
-    var conceptAncestor = TableVariable.forPrimary(TablePointer.fromTableName("concept_ancestor"));
+    var conceptAncestor = TableVariable.forPrimary(TablePointer.fromTableName(CONCEPT_ANCESTOR));
     var conceptIdLiteral = new Literal(conceptId);
+    FieldVariable ancestorConceptId = conceptAncestor.makeFieldVariable(ANCESTOR_CONCEPT_ID);
     return new Query(
-        List.of(conceptAncestor.makeFieldVariable("ancestor_concept_id")),
+        List.of(ancestorConceptId),
         List.of(conceptAncestor),
         BooleanAndOrFilterVariable.and(
             BinaryFilterVariable.equals(
-                conceptAncestor.makeFieldVariable("descendant_concept_id"), conceptIdLiteral),
-            BinaryFilterVariable.notEquals(
-                conceptAncestor.makeFieldVariable("ancestor_concept_id"), conceptIdLiteral)));
+                conceptAncestor.makeFieldVariable(DESCENDANT_CONCEPT_ID), conceptIdLiteral),
+            BinaryFilterVariable.notEquals(ancestorConceptId, conceptIdLiteral)));
+  }
+
+  /**
+   * Generate a subquery that returns true if the outerConcept has any children.
+   *
+   * <pre>{@code
+   * EXISTS (SELECT
+   *     1
+   *   FROM
+   *     concept_ancestor ca
+   *   JOIN
+   *     concept c2
+   *   ON
+   *     c2.concept_id = ca.descendant_concept_id
+   *     AND ca.ancestor_concept_id = c.concept_id
+   *     AND ca.descendant_concept_id != c.concept_id
+   *     AND c2.standard_concept = 'S') AS has_children
+   * }</pre>
+   */
```

Comment:
I personally like having this comment here, and generally think the sql comments make the query code clearer and more approachable. But, I know that we were just talking about how quickly these comments can get out of date, and maybe trying a more durable comment like a description of functionality, and just let tests show the actual expected sql. So, I just wanted to raise that conversation here in case we want changes to be made

Reply:
I think there's value in having them and, once the query code settles down, we shouldn't be changing them as much. But it'd be nice if there was a way to automatically update them when the code changes.